xdsl/utils/lexer.py

```diff
@@ -173,8 +172,42 @@ def from_span(cls, span: Span | None) -> StringLiteral | None:
 
     @property
     def string_contents(self):
-        # TODO: is this a hack-job?
-        return ast.literal_eval(self.text)
+        # Common case, so we optimize for it
+        if "\\" not in self.text:
+            return self.text[1:-1]
+        res = list[str]()
```

Comment:
Is there a reason why `res` is not a string that we append to? Is it so much slower in python?

Reply:
That was my first instinct, I was thinking maybe a mutablebytearray or something

Reply:
Yeah I think that was just some first performance concern. Either way, plan is to have cryptic optimized code here in the end, because it's perf-critical! Just keeping whatever works for a first fix